File: November2021/DailyTemperatures.java
class Solution {
	public int[] dailyTemperatures(int[] temperatures) {
		int len = temperatures.length;
		int[] ans = new int[len];

		Stack<Integer> st = new Stack<>();

		for(int i=0;i<len;i++){

			while(!st.empty() && temperatures[st.peek()] <temperatures[i]){
				//System.out.println(temperatures[st.peek()]);
				// found an increasing value for st.peek() index
				ans[st.peek()] = i - st.peek();
				st.pop();
				// System.out.println( "removed "+ st.peek() + " " + temperatures[st.pop()]);

			}

			st.push(i);

		}
		return ans;
	}
}

File: November2021/C++/Daily_Temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temp) {
        vector<int> stack(temp.size(),0);
        vector<int> res(temp.size(),0);
        int top=-1;
        for(int i=0;i<temp.size();i++)
        {
            while(top>-1 && temp[i]>temp[stack[top]])
            {
                int idx=stack[top--];
                res[idx]=i-idx;
            }
            stack[++top]=i;
        }
        return res;
    }
};